### PR TITLE
Migrate to_subgait_msg to subgait class

### PR DESCRIPTION
**Pull request by Roelemans**
Dependent on https://github.com/project-march/march/pull/411

Closes PM-149

## Description

Migrated to_subgait_msg functionality to the subgait class so that it can be reused by the gait selector.

## Changes

* Migrated to_subgait_msg functionality to the subgait class
* Moved to_joint_trajectory_msg from ModifiableSubgait to Subgait class
* Moved export_to_file to the GaitGeneratorPlugin class

### CODE REVIEW

File: march_rqt_gait_generator/src/march_rqt_gait_generator/gait_generator.py [hunk 1]
```diff
@@ -399,12 +397,49 @@ def export_gait(self):
...
+            if overwrite_file == QMessageBox.No:
+                return
+
+        rospy.loginfo("Writing gait to " + output_file_path)
+
+        try:
+            os.makedirs(output_file_directory)
+        except OSError:
+            if not os.path.isdir(output_file_directory):
+                raise
```
Comment (JorisWeeda) [inline march_rqt_gait_generator/src/march_rqt_gait_generator/gait_generator.py]: a blank raise when catching an exception? this seems like a weird bit of code

File: march_rqt_gait_generator/src/march_rqt_gait_generator/gait_generator.py [hunk 2]
```diff
@@ -399,12 +397,49 @@ def export_gait(self):
...
+
+        output_file = open(output_file_path, 'w')
+        output_file.write(str(subgait_msg))
```
Comment (JorisWeeda) [inline march_rqt_gait_generator/src/march_rqt_gait_generator/gait_generator.py]: i would use with open, this prevents a corrupted file when an error occurs
```suggestion
with open(output_file_path, 'w') as file:
   file.write(str(subgait_msg))
```

File: march_rqt_gait_generator/src/march_rqt_gait_generator/gait_generator.py (hunk 1 above)
Comment (Olavhaasie) [reply march_rqt_gait_generator/src/march_rqt_gait_generator/gait_generator.py]: I think it reraises the `OSError`

File: march_rqt_gait_generator/src/march_rqt_gait_generator/gait_generator.py (hunk 2 above)
Comment (Olavhaasie) [reply march_rqt_gait_generator/src/march_rqt_gait_generator/gait_generator.py]: And don't forget to close it as well
```python
with open(output_file_path, 'w') as file:
    file.write(str(subgait_msg))
    file.close()
```

**Comment by Roelemans** [comment]
> Take a look at M(model) V(view) C(controller) based projects.
Seems like a good idea, though I do noet immediately see how to implement it in this case. In any case, it is not a priority for now I think.

File: march_rqt_gait_generator/src/march_rqt_gait_generator/gait_generator.py (hunk 1 above)
Comment (Roelemans) [reply march_rqt_gait_generator/src/march_rqt_gait_generator/gait_generator.py]: (How) should I improve this? Maybe make a custom exception? (NotAFileException or something)

File: march_rqt_gait_generator/src/march_rqt_gait_generator/gait_generator.py (hunk 2 above)
Comment (Roelemans) [reply march_rqt_gait_generator/src/march_rqt_gait_generator/gait_generator.py]: Excellent suggestion @JorisWeeda. @Olavhaasie closing is not necssary, the file is automatically closed when with exists.

File: march_rqt_gait_generator/src/march_rqt_gait_generator/gait_generator.py (hunk 1 above)
Comment (Olavhaasie) [reply march_rqt_gait_generator/src/march_rqt_gait_generator/gait_generator.py]: I think you should raise the `OSError` always, not only when it is not a dir, you could check for it beforehand

Comment (JorisWeeda) [reply march_rqt_gait_generator/src/march_rqt_gait_generator/gait_generator.py]: i agree, check before hand.

a raise without arguments raises the last stored exception.

Comment (Roelemans) [reply march_rqt_gait_generator/src/march_rqt_gait_generator/gait_generator.py]: Removed try/except construction

**Pull request merged**